pkg/attestlib/public_key.go

```diff
@@ -93,8 +94,9 @@ func extractPgpKeyID(keyData []byte) (string, error) {
 }
 
 func extractPkixKeyID(keyData []byte, keyID string) (string, error) {
-	// TODO(https://github.com/grafeas/kritis/issues/541): Generate ID based on
-	// DER encoding of key when keyID is empty string.
+	if len(keyID) == 0 {
+		return generatePkixPublicKeyId(keyData)
```

Comment:
I would recommend wrapping the error, because users might be confused with why `generatePkixPublicKeyId` in the first place.

Reply:
generatePkixPublicKeyId returns a wrapped or new error that should be descriptive enough for users.